packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase3Algorithm_kokkos_def.hpp

```diff
@@ -94,129 +107,130 @@ namespace MueLu {
+                               aggStat(nodeIdx)         = AGGREGATED;
+                               procWinner(nodeIdx, 0)   = myRank;
+                               vertex2AggId(nodeIdx, 0) = aggId;
+                               // aggregates.SetIsRoot(nodeIdx);
+                               Kokkos::atomic_decrement(&numNonAggregated());
+                               for(int neigh = 0; neigh < neighbors.length; ++neigh) {
+                                 neighIdx = neighbors(neigh);
+                                 if((neighIdx != nodeIdx) &&
+                                    graph.isLocalNeighborVertex(neighIdx) &&
+                                    (aggStatOld(neighIdx) == READY)) {
+                                   aggStat(neighIdx)         = AGGREGATED;
+                                   procWinner(neighIdx, 0)   = myRank;
+                                   vertex2AggId(neighIdx, 0) = aggId;
+                                   Kokkos::atomic_decrement(&numNonAggregated());
+                                 }
+                               }
+                               return;
+                             }
+
+                             // Getting a little desperate!
+                             // Let us try to aggregate into a neighboring aggregate
+                             for(int neigh = 0; neigh < neighbors.length; ++neigh) {
+                               neighIdx = neighbors(neigh);
+                               if (graph.isLocalNeighborVertex(neighIdx) &&
+                                   (aggStatOld(neighIdx) == AGGREGATED)) {
+                                 aggStat(nodeIdx)         = AGGREGATED;
+                                 procWinner(nodeIdx, 0)   = myRank;
+                                 vertex2AggId(nodeIdx, 0) = vertex2AggId(neighIdx, 0);
+                                 Kokkos::atomic_decrement(&numNonAggregated());
+                                 return;
+                               }
+                             }
+
+                             // Getting quite desperate!
+                             // Let us try to make a non contiguous aggregate
+                             if(makeNonAdjAggs) {
+                               for(LO otherNodeIdx = 0; otherNodeIdx < numRows; ++otherNodeIdx) {
+                                 if((otherNodeIdx != nodeIdx) &&
+                                    (aggStatOld(otherNodeIdx) == AGGREGATED)) {
+                                   aggStat(nodeIdx)         = AGGREGATED;
+                                   procWinner(nodeIdx, 0)   = myRank;
+                                   vertex2AggId(nodeIdx, 0) = vertex2AggId(otherNodeIdx, 0);
+                                   Kokkos::atomic_decrement(&numNonAggregated());
+                                   return;
+                                 }
+                               }
+                             }
+
+                             // Total deperation!
```

Comment:
spelling

Reply:
OK, I will fix it in the next wave of refactor as this should not impede the algorithm too much ; )